src/types.jl

```diff
@@ -24,33 +24,37 @@ julia> b[Set(["c"=>"d","a"=>"b"])] == b[Set(["a"=>"b","c"=>"d"])]  # true
 ````
 """
 struct CounterMetric{VT <: Number, LT<: Base.AbstractLock} <: PromMetric
-    name::String                                        # Required for TYPE string
-    desc::String                                        # Description for HELP string, can be blank str
-    label_keys::Vector{String}                          # Ordered List of Possible Keys for use in labels
-    label_data::Dict{Set{Pair{String, String}},VT}      # Dict that stores metric data per label
-    _vtype::Type                                        # for storing the value type, useful for checking
-    _lock::LT                                           # lock
+    name::String                                            # Required for TYPE string
+    desc::String                                            # Description for HELP string, can be blank str
+    label_keys::Vector{String}                              # Ordered List of Possible Keys for use in labels
+    label_data::Dict{Set{Pair{String, String}},VT}          # Dict that stores metric data per label
+    label_timestamp::Dict{Set{Pair{String, String}},Int}    # Dict that stores timestamp data
+    log_timestamp::Bool                                     # Field to check if the label_timestamp field needs to be updated
+    _vtype::Type                                            # for storing the value type, useful for checking
+    _lk::LT                                                 # lock
 end
-CounterMetric(n; vtype::Type=Int) = CounterMetric(n, "", Vector{String}(), Dict{Set{Pair{String, String}}, vtype}(), vtype, Threads.ReentrantLock())
-CounterMetric(n, d; vtype::Type=Int) = CounterMetric(n, d, Vector{String}(), Dict{Set{Pair{String, String}}, vtype}(), vtype, Threads.ReentrantLock())
-CounterMetric(n, d, labels::Vector{String}; vtype::Type=Int) = CounterMetric(n, d, labels, Dict{Set{Pair{String, String}}, vtype}(), vtype, Threads.ReentrantLock())
+CounterMetric(n, t; vtype::Type=Int) = CounterMetric(n, "", Vector{String}(), Dict{Set{Pair{String, String}}, vtype}(), Dict{Set{Pair{String, String}}, Int}(), t, vtype, Threads.ReentrantLock())
```

Comment:
i assume `t` is the boolean for the timestamp? Can you type it, make it a kwarg and put it in the docstrings. It should be kwarg because we want users to explicitly call it when they are enabling/disabling it. Should default to off, because it has performance overhead.